src/main/java/Main.java

```diff
@@ -0,0 +1,64 @@
+import service.Printer;
+import service.PrinterImpl;
+import service.ReportService;
+import service.ReportServiceImpl;
+import service.TransactionStrategy;
+import service.TransactionStrategyImpl;
+import service.transaction.BalanceTransactionHandler;
+import service.transaction.PurchaseTransactionHandler;
+import service.transaction.ReturnTransactionHandler;
+import service.transaction.SupplyTransactionHandler;
+import service.transaction.TransactionHandler;
+
+public class Main {
+    public static final Map<FruitTransaction.Operation,
+                        TransactionHandler> transactionHandlerMap = new HashMap<>();
+    public static final TransactionHandler balanceTransactionHandler =
+                                        new BalanceTransactionHandler();
+    public static final TransactionHandler purchaseTransactionHandler =
+                                        new PurchaseTransactionHandler();
+    public static final TransactionHandler returnTransactionHandler =
+                                        new ReturnTransactionHandler();
+    public static final TransactionHandler supplyTransactionHandler =
+                                        new SupplyTransactionHandler();
+
+    public static void main(String[] args) {
+        transactionHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                                    balanceTransactionHandler);
+        transactionHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                                    purchaseTransactionHandler);
+        transactionHandlerMap.put(FruitTransaction.Operation.RETURN,
+                                    returnTransactionHandler);
+        transactionHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                                    supplyTransactionHandler);
+        String pathToInputDataFile = "src/main/resources/inputData.csv";
+        File inputData = new File(pathToInputDataFile);
+        FileReaderService csvFileReader = new CsvFileReaderService();
+        List<String> dataFromFile = csvFileReader.readFile(inputData);
+        FruitTransactionConverter converter = new FruitTransactionConverterImpl();
+        List<FruitTransaction> fruitTransactions =
+                            converter.convertToFruitTransaction(dataFromFile);
+        TransactionStrategy transactionStrategy =
+                            new TransactionStrategyImpl(transactionHandlerMap);
+        ReportService reportService = new ReportServiceImpl(transactionStrategy);
+        List<String> report = reportService.createReport(fruitTransactions);
+        String pathToReportFile = "src/main/resources/reportFile.csv";
+        File reportFile = new File(pathToReportFile);
+        FileWriterService fileWriterService = new CsvFileWriterService();
+        fileWriterService.saveToFile(reportFile, report);
+        List<String> readReport = csvFileReader.readFile(reportFile);
+        Printer printer = new PrinterImpl();
+        printer.print(readReport);

```

Comment:
Look, you have a plenty of raw logic in Main method. 
Please try to rethink your solution so from Main method you call some service and it's making required magic from just one command like:

fruitShopService.generateDailyReport(INPUT_FILENAME, REPORT_FILENAME);

think of what other services it will require to be initialized, add these required services to be set via constructor

and in the end your Main method will look like:
- Initialize handlers map
- Create instances of required services / dao instances / etc.
- Create your masterService with constructor that will accept lesser services
- Run just 1 command to generate report